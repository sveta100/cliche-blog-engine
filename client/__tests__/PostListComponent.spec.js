import { shallowMount } from '@vue/test-utils';
import PostList from '../src/components/PostListComponent.vue';
import PostTile from '../src/components/PostTileComponent.vue';

jest.mock('axios', () => ({
  delete: (id) => Promise.resolve({
    data: [
      { _id: 1, title: 'blog 1' },
      { _id: 2, title: 'blog 2' }].filter((i) => i !== id),
  }),
}));

describe('PostList component', () => {
  const wrapper = shallowMount(PostList, {
    propsData: {
      items: [],
    },
    mocks: {
      $toasted: {
        error: () => {},
        show: () => {},
      },
    },
  });

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders passed items correctly', async () => {
    wrapper.vm.$options.watch.items.call(wrapper.vm, [
      { _id: 1, title: 'blog 1' },
      { _id: 2, title: 'blog 2' }]);

    wrapper.setProps({
      items: [
        { _id: 1, title: 'blog 1' },
        { _id: 2, title: 'blog 2' }],
    });

    await wrapper.vm.$nextTick();
    expect(wrapper.vm.items.length).toBe(2);
    expect(wrapper.vm.$data.posts.length).toBe(2);
    expect(wrapper.findAllComponents(PostTile).length).toBe(2);
  });

  it('deletes a blog post from the list', async () => {
    wrapper.vm.$root.$emit('deletePost', 2);
    await wrapper.vm.$nextTick();

    expect(wrapper.findAllComponents(PostTile).length).toBe(1);
  });
});
