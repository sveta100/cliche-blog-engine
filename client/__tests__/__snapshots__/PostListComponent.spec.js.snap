// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList component renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
